pkg/ds/daemon_set_test.go

```diff
@@ -82,7 +82,7 @@ func watchDSChanges(
 		defer close(errCh)
 
 		for {
-			var watched dsstore.WatchedDaemonSets
+			var watched *dsstore.WatchedDaemonSets
```

Comment:
I don't understand the rationale for this change. To me, pointers are scary because it's more likely to lead to uncaught runtime errors that crash the program. Can you point to some code elsewhere that benefits from this change? I see only additional error checks in my preliminary scan of this code.